src/expr/src/relation/func.rs

```diff
@@ -749,6 +749,64 @@ fn generate_series_int64(start: Datum, stop: Datum) -> impl Iterator<Item = (Row
     (start..=stop).map(move |i| (Row::pack_slice(&[Datum::Int64(i)]), 1))
 }
 
+fn generate_series_step_int32(
+    start: Datum,
+    stop: Datum,
+    step: Datum,
+) -> impl Iterator<Item = (Row, Diff)> {
+    let start = start.unwrap_int32();
+    let stop = stop.unwrap_int32();
+    let step = step.unwrap_int32();
+
+    let pos_series = start <= stop;
+    let pos_step = step > 0;
+    // todo@jdlaughlin: we should error if step == 0!
+    if step == 0 || (pos_series && !pos_step) || (!pos_series && pos_step) {
```

Comment:
Is there a way to error on specific input values right now? I tried to figure out how to do that in `sql/src/func.rs`, but failed.

Reply:
`sql/src/func.rs` just plans the function so doesn't have access to values (unless you only want to accept literals, which you probably don't), so I think you'd want an `EvalError` here if you think that's appropriate.

Reply:
It would be appropriate! Postgres says:

```
benesch=# select generate_series(1, 10, 0);
ERROR:  step size cannot equal zero
```